config/cmake/HDF5ExampleCache.cmake

```diff
@@ -112,12 +112,35 @@ else ()
     endif ()
   endif ()
   if (HDF5_BUILD_JAVA)
+    # HDF5_JAVA_IMPLEMENTATION is already a CACHE variable set in java/CMakeLists.txt
+    # Use it to determine JNI vs FFM
+    if (HDF5_JAVA_IMPLEMENTATION STREQUAL "JNI")
+      set (HDF5_PROVIDES_JNI TRUE CACHE BOOL "Provides JNI" FORCE)
+      set (HDF5_PROVIDES_COMPAT TRUE CACHE BOOL "Provides Compat" FORCE)
```

Comment:
I disagree on the change made here. If JNI is being built, there shouldn't be any compatbility layer, as JNI is what we've always had up to this point, so there's no need for a compatibility layer. IMO, `HDF5_PROVIDES_(JAVA)_COMPAT` should only ever be set to true when building FFM with the compatibility layer. Having this set to TRUE in both branches here doesn't make sense to me. I think this should go back to FALSE and the more appropriate change is that the variable in hdf5-config.cmake shouldn't always be TRUE (or ON as it currently is)

Reply:
This is examples and refers to the examples that use the H5 class - JNI always uses this class it is optional for FFM. 
I need to allow FFM to not use it.